interfaces/builtin/storage_framework_service_test.go

```diff
@@ -0,0 +1,108 @@
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type StorageFrameworkServiceInterfaceSuite struct {
+	iface       interfaces.Interface
+	coreSlot    *interfaces.Slot
+	classicSlot *interfaces.Slot
+	plug        *interfaces.Plug
+}
+
+var _ = Suite(&StorageFrameworkServiceInterfaceSuite{})
+
+func (s *StorageFrameworkServiceInterfaceSuite) SetUpTest(c *C) {
+	// a storage-framework-service slot on a storage-framework-service snap
+	const storageFrameworkServiceMockCoreSlotSnapInfoYaml = `name: storage-framework-service
+version: 1.0
+apps:
+ app:
+  command: foo
+  slots: [storage-framework-service]
+`
+	// a storage-framework-service slot on the core snap (as automatically added on classic)
+	const storageFrameworkServiceMockClassicSlotSnapInfoYaml = `name: core
+type: os
+slots:
+ storage-framework-service:
+  interface: storage-framework-service
+`
+	const mockPlugSnapInfo = `name: client
+version: 1.0
+apps:
+ app:
+  command: foo
+  plugs: [storage-framework-service]
+`
+	s.iface = &builtin.StorageFrameworkServiceInterface{}
+	// storage-framework-service snap with storage-framework-service slot on an core/all-snap install.
+	snapInfo := snaptest.MockInfo(c, storageFrameworkServiceMockCoreSlotSnapInfoYaml, nil)
+	s.coreSlot = &interfaces.Slot{SlotInfo: snapInfo.Slots["storage-framework-service"]}
+	// storage-framework-service slot on a core snap in a classic install.
+	snapInfo = snaptest.MockInfo(c, storageFrameworkServiceMockClassicSlotSnapInfoYaml, nil)
+	s.classicSlot = &interfaces.Slot{SlotInfo: snapInfo.Slots["storage-framework-service"]}
+
+	plugSnap := snaptest.MockInfo(c, mockPlugSnapInfo, nil)
+	s.plug = &interfaces.Plug{PlugInfo: plugSnap.Plugs["storage-framework-service"]}
+}
+
+func (s *StorageFrameworkServiceInterfaceSuite) TestName(c *C) {
+	c.Check(s.iface.Name(), Equals, "storage-framework-service")
+}
+
+func (s *StorageFrameworkServiceInterfaceSuite) TestSanitizeIncorrectInterface(c *C) {
+	c.Check(func() { s.iface.SanitizeSlot(&interfaces.Slot{SlotInfo: &snap.SlotInfo{Interface: "other"}}) },
+		PanicMatches, `slot is not of interface "storage-framework-service"`)
+	c.Check(func() { s.iface.SanitizePlug(&interfaces.Plug{PlugInfo: &snap.PlugInfo{Interface: "other"}}) },
+		PanicMatches, `plug is not of interface "storage-framework-service"`)
+}
+
+func (s *StorageFrameworkServiceInterfaceSuite) TestUsedSecuritySystems(c *C) {
+	// connected slots have a non-nil security snippet for apparmor
+	apparmorSpec := &apparmor.Specification{}
+	err := apparmorSpec.AddConnectedSlot(s.iface, s.plug, s.coreSlot)
+	c.Assert(err, IsNil)
+	c.Assert(apparmorSpec.SecurityTags(), DeepEquals, []string{"snap.storage-framework-service.app"})
```

Comment:
This will need to be changed to "snap.client.app" due to the requested changes in AppArmorConnectedSlot() (see above).